common/src/main/java/com/wynntils/models/leaderboard/type/LeaderboardProfile.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.models.leaderboard.type;
+
+import com.wynntils.models.profession.type.ProfessionType;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public record LeaderboardProfile(String name, int timePlayed, Map<String, Integer> ranks) {
+    public List<LeaderboardBadge> getBadges() {
```

Comment:
We call this on render time.. Just cache this. (That means making this record a class too)